replication.go

```diff
@@ -2067,57 +2062,79 @@ func (cm *ContentManager) makeDealsForContent(ctx context.Context, content Conte
 			continue
 		}
 
-		// TODO: verify signature!
-		switch dealresp.Response.State {
-		case storagemarket.StorageDealError:
-			if err := cm.recordDealFailure(&DealFailureError{
-				Miner:   ms[i],
-				Phase:   "propose",
-				Message: dealresp.Response.Message,
-				Content: content.ID,
-			}); err != nil {
-				return err
-			}
-		case storagemarket.StorageDealProposalRejected:
-			if err := cm.recordDealFailure(&DealFailureError{
-				Miner:   ms[i],
-				Phase:   "propose",
-				Message: dealresp.Response.Message,
-				Content: content.ID,
-			}); err != nil {
-				return err
-			}
+		propnd, err := cborutil.AsIpld(p.DealProposal)
+		if err != nil {
+			return xerrors.Errorf("failed to compute deal proposal ipld node: %w", err)
+		}
+
+		dealUUID := uuid.New()
+		cd := &contentDeal{
+			Content:  content.ID,
+			PropCid:  util.DbCID{propnd.Cid()},
+			DealUUID: dealUUID.String(),
+			Miner:    ms[i].String(),
+			Verified: verified,
+		}
+
+		if err := cm.DB.Create(cd).Error; err != nil {
```

Comment:
TODO: instead of creating the database row and then deleting it if there's an error, can we just use the deal UUID as an index?

Reply:
Tricky to do because:
- Once an SP running Boost receives and accepts a deal proposal it immediately starts downloading from estuary
- When the download starts, and as it progresses, the shuttle makes RPC calls to the primary estuary node
- The handlers for these RPC calls [need the deal to already have been created in the DB](https://github.com/application-research/estuary/blob/ab982078721d568102186d2aae7bf3c1126b2163/shuttle.go#L302-L313) so they can update its state with transfer progress